File: codes/csharp/008_operator_ternary.cs
using System; // Console, Func<>, Action<>, Action

class Program {
    static void Main(string[] Args) {
        Console.WriteLine("\n// Ternary Operator in C#");

        dynamic CORRECT_ANSWER = "foo";
        Console.WriteLine($"correct answer: \"{CORRECT_ANSWER}\"");

        dynamic MyAnswer;

        MyAnswer = "bar";
        Console.WriteLine($"my answer: \"{MyAnswer}\"");
        Console.WriteLine($"is my answer correct: {((MyAnswer == CORRECT_ANSWER) ? "correct!" : "wrong!")}");
        // is my answer correct: wrong!

        MyAnswer = "foo";
        Console.WriteLine($"my answer: \"{MyAnswer}\"");
        Console.WriteLine($"is my answer correct: {((MyAnswer == CORRECT_ANSWER) ? "correct!" : "wrong!")}");
        // is my answer correct: correct!
    }
}

File: codes/dart/008_operator_ternary.dart
void main() {
    print("\n// Ternary Operator in Dart");

    final CORRECT_ANSWER = "foo";
    print("correct answer: \"${CORRECT_ANSWER}\"");

    dynamic myAnswer;

    myAnswer = "bar";
    print("my answer: \"${myAnswer}\"");
    print("is my answer correct: ${((myAnswer == CORRECT_ANSWER) ? "correct!" : "wrong!")}");
    // is my answer correct: wrong!

    myAnswer = "foo";
    print("my answer: \"${myAnswer}\"");
    print("is my answer correct: ${((myAnswer == CORRECT_ANSWER) ? "correct!" : "wrong!")}");
    // is my answer correct: correct!
}
